examples/cpp/hello_world/cli_options.hpp

```diff
@@ -0,0 +1,75 @@
+#include <iostream>
+
+struct subscriber_config
+{
+    bool use_waitset;
+};
+
+struct hello_world_config
+{
+    std::string entity;
+    subscriber_config sub_config;
+};
+
+
+void print_help()
+{
+    std::cout << "Usage: hello_world [entity] [options]\n"
+              << "Entities:                       Mandatory field\n"
+              << "  publisher\n"
+              << "  subscriber\n"
+              << "Options:                        Optional fields:\n"
+              << "  -w, --waitset                 Use waitset read condition (subscriber only)\n"
+              << "  -h, --help                    Print this help message\n";
```

Comment:
IMO, using one cout and one endl per line looks cleaner:
```suggestion
    std::cout << "Usage: hello_world <entity> [options]"              << std::endl;
    std::cout << ""                                                   << std::endl;
    std::cout << "Entities:"                                          << std::endl;
    std::cout << "  publisher             Run a publisher entity"     << std::endl;
    std::cout << "  subscriber            Run a subscriber entity"    << std::endl;
    std::cout << ""                                                   << std::endl;
    std::cout << "Common options:"                                    << std::endl;
    std::cout << "    -h, --help          Print this help message"    << std::endl;
    std::cout << ""                                                   << std::endl;
    std::cout << "Subscriber options:"                                << std::endl;
    std::cout << "    -w, --waitset       Use waitset read condition" << std::endl;
```